### DISCUSSION TITLE
2 questions, keymapping(or command mapping), and jump between function parameters.

### DISCUSSION THREAD

**Discussion by ws051682** [helix-editor/helix#7314] [Q&A]
can anybody help me on the following 2 issues:
1. I use rust-analyzer as my Language server. it worked. but when auto-complete a function, i don't know how to press to move the cursor between parameters. (like in vscode, it's `Tab`)
2. is there's anyway bind key only to rust? i just want to have ":sh cargo run" to have a hotkey.
i'm struggling for almost 2 days....😂
thanks a lot.

**Comment by masriomarm** [comment]
for the 2nd point, you can have this in your config.toml
```
[keys.normal.space]
c = { r = ":sh cargo run 2>&1 || true" , k = ":sh cargo check 2>&1 || true" , t = ":sh cargo test 2>&1 || true" }
```
As for the 1st point, if you figure it out, kindly let me know :D

**Comment by the-mikedavis** [comment]
For 1: switching between tabstops is not yet implemented.

For 2: keybinds are global so this is not possible. There is an issue discussing it somewhere but I can't find the link at the moment. In general `:sh` is meant for small, short, non-interactive commands rather than building a project. There are clever ways around this though like https://github.com/helix-editor/helix/discussions/6477#discussioncomment-5469031
